libcloud/compute/drivers/kubevirt.py

```diff
@@ -391,63 +772,162 @@ def create_node(
                     )
                     raise KeyError(msg)
 
+                claim_name = disk["volume_spec"]["claim_name"]
+
+                if claim_name not in self.ex_list_persistent_volume_claims(namespace=namespace):
```

Comment:
The method would be a bit more readable if it was refactored into multiple smaller function (e.g. one for creating volume, etc.).